mesonbuild/modules/selinux.py

```diff
@@ -0,0 +1,344 @@
+        KwargInfo('fc_file', (str, File, NoneType)),
+        KwargInfo('install', bool, default=True),
+        KwargInfo('install_dir', (str, NoneType)),
+        KwargInfo('name', (str, NoneType)),
+        KwargInfo('mls', bool, default=True),
+        KwargInfo('type', (str, NoneType)),
+        KwargInfo('distro', (str, NoneType)),
+        KwargInfo('direct_initrc', (bool, NoneType)),
+    )
+    def package(self, state: ModuleState, args: T.Tuple[str], kwargs: TYPE_kwargs) -> ModuleReturnValue:
+        sharedir = os.path.join(state.environment.get_prefix(), state.environment.get_datadir(), 'selinux')
+        tools = self._find_tools(state)
+        for tool_name, tool in tools.items():
+            if not tool or not tool.found():
+                raise MesonException(f'SELinux tool {tool_name} not found')
+
+        package_name = args[0]
+        te_file = self._ensure_file(state, T.cast(T.Union[str, File], kwargs.get('te_file') or f'{package_name}.te'))
+        if_file = self._ensure_file(
+            state, T.cast(T.Union[str, File], kwargs.get('if_file') or f'{package_name}.if'),
+            default_content=f"## <summary>{package_name}</summary>\n")
+        fc_file = self._ensure_file(
+            state, T.cast(T.Union[str, File], kwargs.get('fc_file') or f'{package_name}.fc'),
+            default_content='')
+        install = T.cast(bool, kwargs['install'])
+        install_dir = T.cast(T.Optional[str], kwargs.get('install_dir'))
+        if install and install_dir is None:
+            install_dir = os.path.join(sharedir, 'packages')
+
+        # mimic '/usr/share/selinux/devel/Makefile' behaviour
+        name = T.cast(T.Optional[str], kwargs.get('name'))
+        if name is None:
+            config = '/etc/selinux/config'
+            try:
+                with open(config, 'r', encoding='utf-8') as f:
+                    for line in f:
+                        line = line.strip()
+                        if line.startswith('SELINUXTYPE='):
+                            name = line.split('=', 1)[1].strip().strip('"\'')
+                            break
+            except (IOError, OSError) as e:
+                raise MesonException(f'Failed to read "{config}": {e}')
+
+        mls = T.cast(bool, kwargs.get('mls'))
+        type = T.cast(T.Optional[str], kwargs['type'])
+        if type is None:
+            if name == 'mls':
+                type = 'mls'
+            elif mls:
+                type = 'mcs'
+
+        distro = T.cast(T.Optional[str], kwargs['distro'])
+        direct_initrc = T.cast(T.Optional[bool], kwargs['direct_initrc'])
+
+        headerdir = os.path.join(sharedir, 'devel', 'include')
+        # fallback to system default
+        if not os.path.exists(headerdir):
+            headerdir = '/usr/share/selinux/devel/include'
+        build_conf = os.path.join(headerdir, 'build.conf')
+
+        try:
+            with open(build_conf, 'r', encoding='utf-8') as f:
```

Comment:
If we are opening this file to compute logic in the build process, we also need to mark it as an additional build input so that the ninja backend adds a regenerate_build dependency on the selinux build.conf

Reply:
done